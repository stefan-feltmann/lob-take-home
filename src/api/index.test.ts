import e from 'express'

let request = require('supertest')
let server = require('./index')
jest.setTimeout(50000)

describe('express REST test', () => {
  // let server
  beforeEach(function () {
    // server = require('./index')
  })
  afterAll(function () {
    server.close()
  })
  test('basic', async () => {
    await request(server)
      .get('/')
      .expect(200)
      .then((response) => {
        expect(response.text).toMatchSnapshot()
      })
      .catch((err) => {
        expect(err).toBeNull()
      })
    await request(server)
      .get('/bad')
      .expect(404)
      .then((response) => {
        expect(response.text).toMatchSnapshot()
      })
      .catch((err) => {
        expect(err).toBeNull()
      })
  })
  test('addressesByStreetNumber', async () => {
    await request(server)
      .get('/addressesByStreetNumber?street=1600')
      .expect(200)
      .then((response) => {
        expect(JSON.parse(response.text)).toMatchSnapshot()
      })
      .catch((err) => {
        expect(err).toBeNull()
      })
    await request(server)
      .get('/addressesByStreetNumber?street=Suite 20')
      .expect(200)
      .then((response) => {
        expect(JSON.parse(response.text)).toMatchSnapshot()
      })
      .catch((err) => {
        expect(err).toBeNull()
      })
  })
  test('addressesByState', async () => {
    await request(server)
      .get('/addressesByState?state=MD')
      .expect(200)
      .then((response) => {
        expect(JSON.parse(response.text)).toMatchSnapshot()
      })
      .catch((err) => {
        expect(err).toBeNull()
      })
  })
})
