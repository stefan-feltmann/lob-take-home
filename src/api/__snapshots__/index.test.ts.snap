// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`express REST test addressesByState 1`] = `
Array [
  Object {
    "city": "Baltimore",
    "line1": "3400 N. Charles St.",
    "state": "MD",
    "zip": "21218",
  },
]
`;

exports[`express REST test addressesByStreetNumber 1`] = `
Array [
  Object {
    "city": "San Francisco",
    "line1": "1600 Holloway Ave",
    "state": "CA",
    "zip": "94132",
  },
  Object {
    "city": "San Francisco",
    "line1": "1600 Holloway Ave",
    "line2": "Suite 10",
    "state": "CA",
    "zip": "94132",
  },
  Object {
    "city": "San Francisco",
    "line1": "1600 Holloway Ave",
    "line2": "Suite 20",
    "state": "CA",
    "zip": "94132",
  },
]
`;

exports[`express REST test addressesByStreetNumber 2`] = `
Array [
  Object {
    "city": "San Francisco",
    "line1": "1600 Holloway Ave",
    "line2": "Suite 20",
    "state": "CA",
    "zip": "94132",
  },
]
`;

exports[`express REST test basic 1`] = `""`;

exports[`express REST test basic 2`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /bad</pre>
</body>
</html>
"
`;
